src/state/reducers/rootReducer.js

```diff
@@ -1,33 +1,42 @@
 import initialState from "../store/initialState";
 
 const rootReducer = (state = initialState, action) => {
-  switch (action.type) {
-    case "SET_ARTICLES_INDEX":
-      return {
-        ...state,
-        mainArticles: action.payload,
-      };
+	switch (action.type) {
+		case "SET_ARTICLES_INDEX":
+			return {
+				...state,
+				mainArticles: action.payload,
+			};
 
-    case "SET_SINGLE_ARTICLE":
-      return {
-        ...state,
-        singleArticle: action.payload,
-      };
+		case "SET_SINGLE_ARTICLE":
+			return {
+				...state,
+				singleArticle: action.payload,
+			};
 
-    case "SET_CURRENT_USER":
-      return {
-        ...state,
-        ...action.payload,
-      };
+		case "SET_CURRENT_USER":
+			return {
+				...state,
+				...action.payload,
+			};
 
-    case "SET_ERROR_MESSAGE":
-      return {
-        ...state,
-        errorMessage: action.payload,
-      };
-    default:
-      return state;
-  }
+		case "SET_USER_AS_SUBSCRIBER":
+			return {
+				...state,
+				currentUser: {
+					...state.currentUser,
+					role: action.payload.role,
+				},
+			};
+
+		case "SET_ERROR_MESSAGE":
+			return {
+				...state,
+				errorMessage: action.payload,
+			};
+		default:
```

Comment:
formatting?

Reply:
Have fixed formatter settings and pushed it all back up again, sorry about that!